scripts/organismSpecies/parse_species.py

```diff
@@ -0,0 +1,203 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+'''
+This scirpt will generate data mcf for UniProt Species
+data mcf file.
+
+Run "python3 parse_species.py --help" for usage.
+
+'''
+
+import re
+from absl import app
+from absl import flags
+
+FLAGS = flags.FLAGS
+flags.DEFINE_string('database',
+                    'speclist.txt',
+                    'database file path.',
+                    short_name='f')
+
+flags.DEFINE_string('old_mcf',
+                    'Species.mcf',
+                    'Old species file path.',
+                    short_name='p')
+
+flags.DEFINE_string('output_mcf',
+                    'newSpecies.mcf',
+                    'The output data mcf file path.',
+                    short_name='m')
+
+
+def get_mcf_piece(code, info, seen_name_to_dcid, CODE_TO_KINGDOM):
+    """
+    Args:
+        code: an organism code
+        info: an information map
+    Returns:
+        a data mcf text.
+    Example: code = 'AADNV',
+    info = {'Official': 'Abaeis nicippe', 'kingdom': 'E',
+    'taxonID': '72259', 'Common': 'Sleepy orange butterfly'}
+    """
+    name = get_class_name(info['Official'])
+
+    name_seen = False
+    # if this species is already in Species.mcf in DCs, keep dcid
+    if name in seen_name_to_dcid:
+        dcid = seen_name_to_dcid[name]
+        name_seen = True
+    else:
+        dcid = code
+    type_names = ''
+    if 'Official' in info:
+        type_names += 'scientificName: "' + info['Official'] + '"\n'
+    if 'Common' in info:
+        type_names += 'commonName: "' + info['Common'] + '"\n'
+    if 'Alternate' in info:
+        type_names += 'alternateName: "' + info['Alternate'] + '"\n'
+    taxonID = info['taxonID']
+    if taxonID == "31609":
+        # scientific name = "Parainfluenza virus 5 (isolate Canine/CPI-)"
+        name += 'Negative'
+    elif taxonID == "31608":
+        # scientific name = "Parainfluenza virus 5 (isolate Canine/CPI+)"
+        name += 'Positive'
+    mcf = ('Node: ' + name + '\n' + 'name: "' + name + '"\n' +
+           'typeOf: Species' + '\n' + type_names + 'ncbiTaxonID: "' + taxonID +
+           '"\n' + 'organismTaxonomicKingdom: dcs:' +
+           CODE_TO_KINGDOM[info['kingdom']] + '\n' + 'uniProtOrganismCode: "' +
+           code + '"\n' + 'dcid: "bio/' + dcid + '"\n')
+
```

Comment:
nit: this may be more readable if you collect lines and join.

```
mcf_lines = []
mcf_lines.append('Node: ' + name)
mcf_lines.append('name: "' + name '"')
...
return '\n'.join(mcf_lines), name_seen
```

Reply:
Nice suggestion! Changed.